src/render.rs

```diff
@@ -232,72 +211,24 @@ impl Render for Filter {
         py: Python<'py>,
         template: TemplateString<'t>,
         context: &mut Context,
-    ) -> Result<Option<Content<'t, 'py>>, PyRenderError> {
+    ) -> TemplateResult<'t, 'py> {
         let left = self.left.resolve(py, template, context)?;
-        Ok(match &self.filter {
-            FilterType::Add(right) => {
-                let left = match left {
-                    Some(left) => left,
-                    None => return Ok(None),
-                };
-                let right = right
-                    .resolve(py, template, context)?
-                    .expect("missing argument in context should already have raised");
-                match (left.to_bigint(), right.to_bigint()) {
-                    (Some(left), Some(right)) => return Ok(Some(Content::Int(left + right))),
-                    _ => {
-                        let left = left.to_py(py);
-                        let right = right.to_py(py);
-                        match left.add(right) {
-                            Ok(sum) => return Ok(Some(Content::Py(sum))),
-                            Err(_) => return Ok(None),
-                        }
-                    }
-                }
+        let result = match &self.filter {
+            FilterType::Add(right, AddFilter) => {
+                AddFilter::apply(left, right, py, template, context)
             }
```

Comment:
With `ResolveFilter` this would become:

```suggestion
            FilterType::Add(filter) => filter.resolve(py, template, context, left),
```